### PR TITLE
hardcaml.1.1, hardcaml-waveterm.0.1, hardcaml-vpi.0.1, hardcaml-examples...

### PR DESCRIPTION
**Pull request by andrewray**
....0.1

### CODE REVIEW

**Comment by avsm** [comment]
!

File: packages/hardcaml/hardcaml.1.1.0/opam [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+]
+remove: [
+  [make "uninstall"]
+]
+depends: [
+  "ocamlfind"
+  "base-bytes"
+  "camlp4"
+  "js_of_ocaml"
+  "ctypes"
```
Comment (avsm) [inline packages/hardcaml/hardcaml.1.1.0/opam]: do you have a minimum version of ctypes here? the interface has changed slightly in 0.3 and 0.4

**Comment by andrewray** [comment]
Seems to be working...disabling hardcaml-vpi on osx as I don't have a clue how to link the thing.  ujamjar/hardcaml-vpi#1

**Comment by andrewray** [comment]
Oh dear, I've broken all the opam 1.1 runs with the new `available: []` syntax.  Do I need to fix this?

**Comment by avsm** [comment]
@AltGr -- any idea what's up with the 1.1 translation script here?

**Comment by AltGr** [comment]
`opam-version:` is the version of the OPAM file format, it should be `1.2` (which will trigger the rewrite for OPAM 1.1). The package version itself can be specified in a `version:` field, but in the repository the directory name is sufficient.

`opam lint` should help track this kind of mistakes :)

**Comment by avsm** [comment]
`opam lint` eh. sounds like something that should run in `.travis-ci.sh`... is it safe to do so before any builds?

**Comment by avsm** [comment]
restarted a failed build, looks very likely that it'll pass

**Comment by AltGr** [comment]
@avsm: definitely, I plan to add it

**Pull request merged**